### PR TITLE
My Museum Tour search pagination

### PR DESCRIPTION
**Pull request by zachgarwood**

### CODE REVIEW

File: src/hooks/useFetch.jsx [hunk 1]
```diff
@@ -12,19 +14,27 @@ import { useEffect, useState } from "react";
...
   const [abortController, setAbortController] = useState(null);
-  const { dataSubSelector, dataSetter, fetchingSetter, errorSetter } =
-    options || {};
+  const {
+    dataSubSelector,
+    dataSetter,
+    paginationSelector,
+    paginationSetter,
+    fetchingSetter,
+    errorSetter,
```
Comment (zachgarwood) [inline src/hooks/useFetch.jsx:31]: I'm not entirely sure why these setters need to be passed in when they are also created above via `useState`:
```js
const [pagination, setPagination] = useState(null);
```

File: src/components/search/Pagination.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React, { useContext } from "react";
+import { SearchContext } from "../../contexts/SearchContext";
+import PageNumber from "./PageNumber";
+
+function Pagination() {
+  const { pagination } = useContext(SearchContext);
+  const pages = Array.from(
+    Array(pagination?.total_pages ?? 1),
+    (_, index) => index + 1,
+  );
```
Comment (zachgarwood) [inline src/components/search/Pagination.jsx:10]: Javascript doesn't have a `range()` function like PHP, so we have to do this convoluted song and dance.

File: src/hooks/useFetch.jsx (hunk 1 above)
Comment (zachgarwood) [reply src/hooks/useFetch.jsx]: I'm not sure what the original reason was for implementing `useFetch` this way, but it was interfering with testing, so I made `useFetch` retrieve state directly from the `SearchContext`.

File: src/components/search/Pagination.jsx (hunk 2 above)
Comment (zachgarwood) [reply src/components/search/Pagination.jsx]: I moved this into the `utils.js` file.

**Review by web-dev-trev — APPROVED** [review]
👍

**Pull request merged**